src/Symfony/Component/DomCrawler/Form.php

```diff
@@ -21,19 +21,19 @@
  */
 class Form extends Link implements \ArrayAccess
 {
-    private \DOMElement $button;
-    private FormFieldRegistry $fields;
-    private ?string $baseHref;
+    private \DOMElement|\DOM\Element $button;
+    protected FormFieldRegistry $fields;
+    protected ?string $baseHref;
 
     /**
-     * @param \DOMElement $node       A \DOMElement instance
-     * @param string|null $currentUri The URI of the page where the form is embedded
-     * @param string|null $method     The method to use for the link (if null, it defaults to the method defined by the form)
-     * @param string|null $baseHref   The URI of the <base> used for relative links, but not for empty action
+     * @param \DOMElement|\DOM\Element $node       A \DOMElement instance
+     * @param string|null              $currentUri The URI of the page where the form is embedded
+     * @param string|null              $method     The method to use for the link (if null, it defaults to the method defined by the form)
+     * @param string|null              $baseHref   The URI of the <base> used for relative links, but not for empty action
      *
      * @throws \LogicException if the node is not a button inside a form tag
      */
-    public function __construct(\DOMElement $node, ?string $currentUri = null, ?string $method = null, ?string $baseHref = null)
+    public function __construct(mixed $node, ?string $currentUri = null, ?string $method = null, ?string $baseHref = null)
```

Comment:
this should use a union type instead of `mixed`

Reply:
This class is not internal as well. Isn't it a BC break if someone extends it actually?